test/unit/ReservoirPair.t.sol

```diff
@@ -206,4 +207,35 @@ contract ReservoirPairTest is BaseTest {
         uint256 lNewShares = _pair.balanceOf(address(_platformFeeTo)) - lPlatformShares;
         assertLt(lNewShares, lPlatformShares);
     }
+
+    function testManipulateOracle_TwoSwapsInOneBlock() external allPairs {
+        // arrange
+        _stepTime(12);
+        uint256 lAmtToSwap = Constants.INITIAL_MINT_AMOUNT * 3;
+
+        // act - since the first swap is the one that gets registered, we manipulate the price by several folds
+        _tokenA.mint(address(_pair), lAmtToSwap);
+        uint256 lAmtOut = _pair.swap(int256(lAmtToSwap), true, address(this), "");
+
+        (uint256 lReserve0, uint256 lReserve1,,uint16 lIndex) = _pair.getReserves();
+
+        // immediately arb back
+        _tokenB.transfer(address(_pair), lAmtOut);
+        _pair.swap(-int256(lAmtOut), true, address(this), "");
+
+        (lReserve0, lReserve1,, lIndex) = _pair.getReserves();
+
+        console.log(lReserve0);
+        console.log(lReserve1);
+
+        Observation memory lObs = _oracleCaller.observation(_pair, lIndex);
+
+        console.log(LogCompression.fromLowResLog(lObs.logInstantRawPrice));
+        console.log(LogCompression.fromLowResLog(lObs.logInstantClampedPrice));
+
+        // Attacker could repeat this for every second by only paying the swap fees. Thw mitigation factors on our end are that:
+        // 1. Arbitrage bots could submit txs that get sandwiched between the two malicious txs. How realistic is this if the attacker broadcasts 2 txs simultaneously (or even better, make it atomic via a contract call)?
+        //    - seems quite unlikely to me, cuz by the time the the block is included, the price would not have changed much to the arb bots, unless they are really sophisticated
+        // 2. The effect of the clamp
```

Comment:
What does this attack achieve? Every time the attacker trades back to normal price they update the accumulator with the correct price. So they're actually just making the oracle more accurate? The instant price only affects for 1 second and if we clamp the rate of change per second that shouldn't matter much at all

Reply:
Nope, because we only update the 1st trade of the timestamp, when they trade / arb back to the normal price in the same block, nothing happens to that oracle reading in the 2nd trade of the block. 

**In the same block**
1st trade - manipulate oracle, instant prices affected 
2nd trade - arb it back so no one else does it

Monitor the pair to make sure no other trades take place: that way the instant prices get time to accumulate. If someone comes in to trade, the instant prices get yanked towards fair price. The attacker can repeat the same 2-trade action to mess with the instant prices again, and the monitor. 

To the outside world monitoring prices, it didn't change much. But sitting in the oracle sample is the manipulated instant prices waiting to be written and accumulated.

Reply:
Instant prices don't get written into the oracle but you mean because the oracle price accumulates weight until the next trade it can influence the oracle up the max trade amount?

Reply:
Feels like cleanest solution is to just write the oracle every swap?

Reply:
Yes the cleanest is to write the oracle every swap, and only updating the instant price if it's not the first swap of the block, which is the previous impl with the `else` branch. 

This is the only way to force any manipulators to put their capital at risk (increase the cost) when moving prices, as there is no guarantee they can arb it back in the next block. And if they arb it back in the next block, then the manipulation only lasted for 1 second

Reply:
Why the need for an else if you write the update on every swap, because otherwise it will move the index?

Reply:
yes, then we'll have multiple samples per timestamp which creates problems when reading the oracle.

The alternative is to read if `lPrevious.timestamp` is the current timestamp we're attempting to write, and if so, only overwrite the instant prices. But that's basically an `else` as well